xla/service/gpu/transforms/cudnn_fused_conv_rewriter.cc

```diff
@@ -488,120 +527,217 @@ void CaptureConvGraphRecursive(HloInstruction* instr,
-        num_linear_users++;
+    if (Match(user, m::MultiplyAnyOrder(&op0, m::Op().Is(instr),
+                                        m::Broadcast(m::Op(&operand0)))) &&
+        ShapeUtil::IsScalar(operand0->shape())) {
+      if (graph_string.AppendOp("scale", op0, {instr, operand0})) {
+        operands.push_back(operand0);
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // Inverse Scale
-    if (Match(user, m::Divide(&op, m::Op(&operand0),
-                              m::Broadcast(m::Op(&operand1)))) &&
-        ShapeUtil::IsScalar(operand1->shape())) {
-      if (graph_string.AppendOp("invscale", op, {operand0, operand1})) {
-        operands.push_back(operand1);
-        num_linear_users++;
+    if (Match(user, m::Divide(&op0, m::Op().Is(instr),
+                              m::Broadcast(m::Op(&operand0)))) &&
+        ShapeUtil::IsScalar(operand0->shape())) {
+      if (graph_string.AppendOp("invscale", op0, {instr, operand0})) {
+        operands.push_back(operand0);
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // ReLU
-    if (Match(user, m::MaximumAnyOrder(&op, m::Op(&operand0),
+    if (Match(user, m::MaximumAnyOrder(&op0, m::Op().Is(instr),
                                        m::Broadcast(m::ConstantScalar(0))))) {
-      if (graph_string.AppendOp("relu", op, {operand0})) {
-        num_linear_users++;
+      if (graph_string.AppendOp("relu", op0, {instr})) {
+        ++num_new_users;
+        CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
+                                  visited_instrs, final_instr, num_endpoints);
+      }
+      continue;
+    }
+    //  Maximum of the absolute value (Amax) following ReLU (elided Abs)
+    if (Match(user, m::Reduce(&op0, m::Op().Is(instr), m::Op())) &&
+        graph_string.OpInGraph(instr, "relu") && AppliesMaxReduce(op0)) {
+      if (graph_string.AppendOp("amax", op0, {instr})) {
+        aux_outputs.push_back(op0);
+        ++num_new_users;
+      }
+      continue;
+    }
+    // ReLU6, represented in the cuDNN graph as min(ReLU, 6)
+    if (Match(
+            user,
+            m::Clamp(
+                &op1,
+                m::Broadcast(&op0, m::ConstantEffectiveScalar(0)).WithOneUser(),
+                m::Op().Is(instr),
+                m::Broadcast(&operand0, m::ConstantEffectiveScalar(6))
+                    .WithOneUser()))) {
+      if (!graph_string.OpInGraph(op0) && !graph_string.OpInGraph(op1)) {
+        graph_string.AppendOp("relu", op0, {instr});
```

Comment:
You never addressed comment

> As discussed offline, this "hack" of having cudnn ops that don't correspond to HLO ops should be commented

Should have a comment stating its intentional the cudnn ops are different from the HLO ops, but that it's fine.